datadriven/src/line_sparser.rs

```diff
@@ -0,0 +1,136 @@
+use crate::test_data::CmdArg;
+use anyhow::Result;
+use regex::Regex;
+
+// Token
+// (1) argument (no value)
+// (2) argument= (no value)
+// (3) argument=() (no value)
+// (4) argument=a (single value)
+// (5) argument=a,b,c (multiple value)
+// (6) argument=(a,b,c,...) (multiple value)
+//
+// parse_line parses a line of datadriven input language and returns
+// the parsed command and CmdArgs.
+pub fn parse_line(line: &str, logger: &slog::Logger) -> Result<(String, Vec<CmdArg>)> {
+    let fields = split_directives(line)?;
+    if fields.is_empty() {
+        bail!("empty lines occurs, unexpected.");
+    }
+
+    debug!(logger, "argument after split: {:?}", fields);
+
+    let cmd = fields[0].clone();
+    let mut cmd_args = vec![];
+
+    for arg in &fields[1..] {
+        let key_value = arg
+            .replacen("=", " ", 1)
+            .split_terminator(' ')
+            .map(|v| v.to_string())
+            .collect::<Vec<String>>();
+        match key_value.len() {
+            1 => {
+                // key only
+                cmd_args.push(CmdArg {
+                    key: key_value[0].clone(),
+                    vals: vec![],
+                })
+            }
+            2 => {
+                let (key, mut val) = (key_value[0].clone(), key_value[1].clone());
+
+                let vals: Vec<String>;
+
+                if val.starts_with('(') && val.ends_with(')') {
+                    // remove ')'
+                    val.pop();
+                    // remove '('
+                    val.remove(0);
+                    // trim because white space is allow.
+                    vals = val
+                        .split_terminator(',')
+                        .map(|v| v.trim().to_string())
+                        .collect();
+                } else {
+                    vals = val.split_terminator(',').map(|v| v.to_string()).collect();
```

Comment:
Let's treat "a,b,c" as single value to keep it simple.

Reply:
Do you mean changing `vals` to String instead of Vec<String>?

Reply:
What I mean is
```
// (1) argument (no value)
// (2) argument= (no value)
// (3) argument=() (no value)
// (4) argument=a (single value)
// (5) argument=a,b,c (single value)
// (6) argument=(a,b,c,...) (multiple value)
```

Reply:
when computing `sum` a=1,2,3, we will get "1,2,3" as value, while if we get `sum` b=(2,3,4), we will get ["2","3","4"].
We will need to add additional code to consider which types are they.
Is this expected?

Reply:
The former, we need to split with a comma and parse the number, the latter we just need to parse the number

Reply:
I'm not sure if I understand you completely. If users want to use multiple values,

Reply:
Actually I am not more in favor of either one. The principle is following the upstream convention.